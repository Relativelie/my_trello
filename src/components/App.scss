@import "../templates";


.listsBlock {
    display: flex;
    height: calc(100vh - ($addListMargin + $addListHeight));
}