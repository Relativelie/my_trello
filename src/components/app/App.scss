@import "../../templates";


.listsBlock {
    display: flex;
    width: 110%;
}

.listsBlock__container {
    display: flex;
    height: calc(100vh - ($addListMargin + $addListHeight + $menuHeight));
}
