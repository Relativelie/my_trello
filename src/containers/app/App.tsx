import { Menu } from '@components/index';
import TaskBoard from '@containers/TaskBoard';

export default function App() {
  return (
    <>
      <Menu />
      <TaskBoard />
    </>
  );
}
