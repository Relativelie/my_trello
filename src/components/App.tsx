import { Children } from 'react';
import { DragDropContext, Droppable, DropResult } from 'react-beautiful-dnd';

import { useTypedSelector } from '../hooks/useTypedSelector';
import { useActions } from '../hooks/useActions';

import { AddList } from './addList/AddList';
import { Lists } from './lists/Lists';
import { Tasks } from './tasks/Tasks';
import { Menu } from './menu/Menu';

import './App.scss';

export default function App() {
    const { dragDropList,
        dragDropListWithTasks,
        dragDropTasks,
        addNewList,
        addTaskArrayToNewList } = useActions();
    const { lists } = useTypedSelector((state) => state.listOfTasksReducer);
    const { tasks } = useTypedSelector((tasksState) => tasksState.tasksReducer);

    const handleOnDragEnd = (e: DropResult) => {
        if (e.destination !== undefined && e.destination !== null) {
            const indexTo = e.destination.index;
            const indexFrom = e.source.index;
            if (e.type === 'lists') {
                dragDropList(indexTo, indexFrom);
                dragDropListWithTasks(indexTo, indexFrom);
            } else {
                const listFrom = getValueFromStr(e.source.droppableId, 9);
                const listTo = getValueFromStr(e.destination.droppableId, 9);
                dragDropTasks(listTo, listFrom, indexTo, indexFrom);
            }
        }
    };

    const getValueFromStr = (value: string, countOfStr: number) => {
        return parseInt(value.substring(value.length, countOfStr), 10);
    };

    const listsAdding = () => {
        addNewList();
        addTaskArrayToNewList(lists.length);
    };

    return (
        <div>
            <Menu />
            <main>
                <AddList listsAdding={listsAdding} />
                <DragDropContext onDragEnd={(e) => handleOnDragEnd(e)}>
                    <Droppable droppableId="lists" type="lists" direction="horizontal">
                        {(provided) => (
                            <div className="listsBlock" {...provided.droppableProps} ref={provided.innerRef}>
                                <div className="listsBlock_container">
                                    {lists.map((list: string, listIndex: number) => (
                                        <Lists key={listIndex} list={list} index={listIndex}>
                                            {Children.map(tasks[listIndex], (child, taskIndex) => (
                                                <Tasks
                                                    listIndex={listIndex}
                                                    taskIndex={taskIndex}
                                                    task={child}
                                                />
                                            ))}
                                        </Lists>
                                    ))}
                                </div>
                                {provided.placeholder}
                            </div>
                        )}
                    </Droppable>
                </DragDropContext>
            </main>
        </div>
    );
}
