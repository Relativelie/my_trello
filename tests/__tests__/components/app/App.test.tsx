import App from "../../../../src/containers/app/App";
import { render } from "../../../testsSetup/test-utils";

describe("App component", () => {
  test("snapshot - app component", () => {
    const tree = render(<App />);
    expect(tree).toMatchSnapshot();
  });
});
