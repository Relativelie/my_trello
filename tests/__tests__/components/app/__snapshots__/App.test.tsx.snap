// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot - app component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="menu"
        >
          <div
            class="menu__container"
          >
            <h3
              class="menu__header"
            >
              World of GuRepio planners
            </h3>
            <ul
              class="menu__elements"
            >
              <li
                class="menu__elements__item"
              >
                There could be a menu
              </li>
              <li
                class="menu__elements__item"
              >
                <address>
                  <a
                    class="menu__elements__item menu__elements__anchor"
                    href="https://github.com/Relativelie?tab=repositories"
                    rel="noreferrer"
                    target="_blank"
                  >
                    Go to github
                  </a>
                </address>
              </li>
            </ul>
          </div>
        </nav>
        <main>
          <div
            class="addList__buttonContainer"
          >
            <button
              type="button"
            >
              Add another list
            </button>
          </div>
          <div
            class="listsBlock"
            data-rbd-droppable-context-id="0"
            data-rbd-droppable-id="lists"
          >
            <div
              class="listsBlock__container"
            />
          </div>
        </main>
      </div>
    </div>
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-0"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      id="rbd-hidden-text-0-hidden-text-0"
      style="display: none;"
    >
      
  Press space bar to start a drag.
  When dragging you can use the arrow keys to move the item around and escape to cancel.
  Some screen readers may require you to be in focus mode or to use your pass through key

    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="menu"
      >
        <div
          class="menu__container"
        >
          <h3
            class="menu__header"
          >
            World of GuRepio planners
          </h3>
          <ul
            class="menu__elements"
          >
            <li
              class="menu__elements__item"
            >
              There could be a menu
            </li>
            <li
              class="menu__elements__item"
            >
              <address>
                <a
                  class="menu__elements__item menu__elements__anchor"
                  href="https://github.com/Relativelie?tab=repositories"
                  rel="noreferrer"
                  target="_blank"
                >
                  Go to github
                </a>
              </address>
            </li>
          </ul>
        </div>
      </nav>
      <main>
        <div
          class="addList__buttonContainer"
        >
          <button
            type="button"
          >
            Add another list
          </button>
        </div>
        <div
          class="listsBlock"
          data-rbd-droppable-context-id="0"
          data-rbd-droppable-id="lists"
        >
          <div
            class="listsBlock__container"
          />
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
